index.d.ts

```diff
@@ -10,226 +10,233 @@ declare global {
+   */
+  getWalletNames(): Promise<{ [addr: string]: string }>;
+
+  /**
+   * Sign a transaction.
+   *
+   * @param transaction A valid Arweave transaction without a wallet keyfile added to it
+   * @param options Arweave signing options
+   *
+   * @returns Promise of a signed transaction instance
+   */
+  sign(
+    transaction: Transaction,
+    options?: SignatureOptions
+  ): Promise<Transaction>;
+
+  /**
+   * Get the permissions allowed for you site by the user.
+   *
+   * @returns Promise of a list of permissions allowed for your dApp.
+   */
+  getPermissions(): Promise<PermissionType[]>;
+
+  /**
+   * Encrypt a string, using the user's wallet.
+   *
+   * @param data Buffer-like structure to encrypt.
+   * @param options Encrypt options. See https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/encrypt#algorithm.
+   *
+   * @returns Promise of the encrypted `Uint8Array`.
+   */
+  encrypt(
+    data: BinaryDataType,
+    options?: RsaOaepParams | AesCtrParams | AesCbcParams | AesGcmParams,
+  ): Promise<Uint8Array>;
+
+  /**
+   * Decrypt a string encrypted with the user's wallet.
+   *
+   * @param data Buffer-like structure to decrypt.
+   * @param options Decrypt options. See https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/decrypt#algorithm.
+   *
+   * @returns Promise of the decrypted `Uint8Array`.
+   */
+  decrypt(
+    data: BinaryDataType,
+    options?: RsaOaepParams | AesCtrParams | AesCbcParams | AesGcmParams,
+  ): Promise<Uint8Array>;
+
+  /**
+   * Get the user's custom Arweave config set in the extension
+   *
+   * @returns Promise of the user's Arweave config
+   */
+  getArweaveConfig(): Promise<GatewayConfig>;
+
+  /**
+   * @deprecated Find alternatives at https://docs.arconnect.io/api/signature
+   */
+  signature(
+    data: BinaryDataType,
+    // https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/sign#algorithm
+    algorithm?: RsaPssParams | EcdsaParams | { name: "RSASSA-PKCS1-v1_5" } | { name: "HMAC" },
+  ): Promise<Uint8Array>;
+
+  /**
+   * Get the user's active public key, from their wallet
+   *
+   * @returns Promise of the active public key
+   */
+  getActivePublicKey(): Promise<string>;
+
+  /**
+   * Add a token to ArConnect (if it is not already present)
+   *
+   * @param id Token contract ID
+   * @param type Type of the token (asset or collectible)
+   * @param gateway Gateway config for the token
+   */
+  addToken(id: string, type?: TokenType, gateway?: GatewayConfig): Promise<void>;
+
+  /**
+   * Checks if a token has been added to ArConnect
+   * 
+   * @param id Token to check for
+   * 
+   * @returns Token added or not
+   */
+  isTokenAdded(id: string): Promise<boolean>;
+
+  /**
+   * Dispatch an Arweave transaction (preferably bundled)
+   *
+   * @param transaction Transaction to dispatch
+   * 
+   * @returns Dispatched transaction ID and type
+   */
+  dispatch(transaction: Transaction): Promise<DispatchResult>;
```

Comment:
The docs here say:

> The function returns the [result](https://docs.arconnect.io/api/dispatch#dispatch-result) of the API call.

But `DispatchResult` only includes two properties, not the whole result. Additionally, when uploading to Turbo, it returns:

```
    return {
      arConfetti: await arconfettiIcon(),
      res: {
        id: dataEntry.id,
        type: "BUNDLED"
      }
    };
```

I think it should not be `dataEntry.id`, which is set by `dataEntry.sign(dataSigner)`, but the response from Turbo that includes the transaction id (`uploadDataToTurbo()`).